hazelcast-mesos-scheduler/src/main/java/com/hazelcast/mesos/util/HazelcastProperties.java

```diff
@@ -1,11 +1,15 @@
 package com.hazelcast.mesos.util;
 
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
 import static com.hazelcast.mesos.util.Util.option;
 import static java.lang.Integer.parseInt;
 
 public class HazelcastProperties {
     static String HAZELCAST_VERSION = option("HAZELCAST_VERSION").or("3.6");
-    static String HOST = option("HOST").or("localhost");
+    static String HOST = option("HOST").or("MESOS_HOSTNAME");
```

Comment:
Yes, MESOS_HOSTNAME provided by mesos when you are starting your mesos_agent with parameter `--hostname`. MESOS_HOSTNAME is the same as using `--hostname`. Better explanation can be found below :
"--hostname=VALUE  :	The hostname the agent node should report, or that the master should advertise in ZooKeeper. If left unset, the hostname is resolved from the IP address that the master/agent binds to"